pyproject.toml

```diff
@@ -0,0 +1,3 @@
+# When changing the version numbers here, also adjust them in `setup.py`
+[build-system]
+requires = ["setuptools", "wheel", "cython >= 0.28.2", "numpy >= 1.12.0"]
```

Comment:
I think we need a fixed old numpy version, depending on the python version. 
At least it's what we did last time (https://github.com/pandas-dev/pandas/commit/77bfe21c7229e724d01721bb84861283baf7e9d3, and it is also what scipy did before they removed it again).
This is because the build happens in an isolated build environment (and with above syntax will always use the latest version), so if your actually installed numpy is then older, it gives problems.

Reply:
Ah, I'll add that to the `toml`.

Reply:
Actually this fails in my tests as `numpy==1.12.1` doesn't compile on Alpine.

Reply:
> Actually this fails in my tests as `numpy==1.12.1` doesn't compile on Alpine.

Is that a known issue with numpy? 
But not really sure how to deal with that. I think we need the pins here for the general case. It might be, that if our setup does not work for a specific use case / platform, you will need to use a `--no-build-isolation`

Reply:
Yes, this is a known issue with NumPy and was fixed in the 1.13 series. I have now pinned the Python 3.6 build also to 1.13.1 and added an explanatory comment. As Alpine+Python3.6 was the initial origin for tackling the `pyproject.toml` again, I made this change.